pkg/graveler/graveler.go

```diff
@@ -985,18 +985,19 @@ func (g *KVGraveler) UpdateBranch(ctx context.Context, repositoryID RepositoryID
 	}
 	newBranch := &Branch{}
 
-	err = g.RefManager.BranchUpdate(ctx, repositoryID, branchID, func(b *Branch) (*Branch, error) {
-		newBranch.CommitID = reference.CommitID
-		newBranch.StagingToken = b.StagingToken
-
+	err = g.RefManager.BranchUpdate(ctx, repositoryID, branchID, func(curBranch *Branch) (*Branch, error) {
 		// TODO(Guys) return error only on conflicts, currently returns error for any changes on staging
-		empty, err := g.stagingEmpty(ctx, repositoryID, repo, b)
+		empty, err := g.stagingEmpty(ctx, repositoryID, repo, curBranch)
 		if err != nil {
 			return nil, err
 		}
 		if !empty {
 			return nil, ErrConflictFound
 		}
+
+		newBranch.CommitID = reference.CommitID
```

Comment:
Is there a way to perform deep copy? This can potentially save us bugs and maintenance of this code